File: tests/fixtures/v2-deserialization-flow/positive/deserialization-flow.js
function direct(req, yaml) {
  return yaml.load(req.body.payload)
}

function propagated(req, yaml) {
  const serialized = req.body.payload
  const alias = serialized
  return yaml.load(alias)
}

function restricted(req, yaml) {
  return yaml.load(req.body.restricted, { schema: yaml.JSON_SCHEMA })
}

File: tests/fixtures/v2-deserialization-flow/positive/deserialization_flow.go
package deserializationflow

func direct(r *http.Request) error {
	var target any
	return gob.NewDecoder(strings.NewReader(r.FormValue("payload"))).Decode(&target)
}

func propagated(r *http.Request) error {
	serialized := r.FormValue("payload")
	alias := serialized
	var target any
	return gob.NewDecoder(strings.NewReader(alias)).Decode(&target)
}

func restricted(r *http.Request) error {
	return json.Unmarshal([]byte(r.FormValue("restricted")), &UploadDTO{})
}
